components/Headers/Alarm.tsx

```diff
@@ -1,28 +1,39 @@
 import Image from 'next/image';
+import { useState } from 'react';
+import NotificationWrapper from '@/components/Notification/NotificationWrapper';
 
 interface AlarmProps {
   isLoggedIn: boolean;
 }
 
-/**
- * 알림 컴포넌트
- * @param isLoggedIn 로그인 여부 판별
- */
-
 export default function Alarm({ isLoggedIn }: AlarmProps) {
+  const [isOpen, setIsOpen] = useState(false);
+  const handleAlarmOpen = () => {
+    if (isOpen) {
+      setIsOpen(false);
+      return;
+    }
+    setIsOpen(true);
+  };
+  const handleAlarmClose = () => setIsOpen(false);
+
   if (!isLoggedIn) {
     return null;
   }
 
   return (
-    <button>
-      <Image
-        src="/icon/icon-alarm.svg"
-        className="mo:hidden"
-        alt="알림 아이콘"
-        width={32}
-        height={32}
-      />
-    </button>
+    <div className="relative">
+      <button onClick={handleAlarmOpen}>
```

Comment:
html 구조상 button 태그 안에 button 태그를 중복으로 사용이 불가합니다!
`In HTML, <button> cannot be a descendant of <button>.
This will cause a hydration error.` 에러 발생합니다

해당 부분
```suggestion
<div role="button" tabIndex={0} onClick={handleAlarmOpen}>
```
로 변경하면 좋을 듯 해요!
그리고 알람 클릭 후 알람 모달이 떳을때, 종 모양 한번더 클릭하면 모달이 닫히는걸 예상했는데 닫히지않고 다시 열리네요! 해당 액션 개선되면 좋을것같습니다!

Reply:
수정했습니다 확인 부탁드립니다!

Reply:
디코 자유대화방 봐주세요 ㅠㅠ